Modules/nismodule.c

```diff
@@ -213,7 +213,9 @@ nis_cat (PyObject *self, PyObject *args, PyObject *kwdict)
     dict = PyDict_New ();
     if (dict == NULL)
         return NULL;
+    #pragma GCC diagnostic ignored "-Wcast-function-type"
```

Comment:
This needs a preprocessor guard to ensure the pragma is only used with compatible compilers.